diff --git a/patches/libwebrtc.patch b/patches/libwebrtc.patch
new file mode 100644
index 0000000..7da49a9
--- /dev/null
+++ b/patches/libwebrtc.patch
@@ -0,0 +1,13 @@
+diff --git a/BUILD.gn b/BUILD.gn
+index e60d7dd0bd..b9b6acab8b 100644
+--- a/BUILD.gn
++++ b/BUILD.gn
+@@ -29,7 +29,7 @@ if (!build_with_chromium) {
+   # 'ninja default' and then 'ninja all', the second build should do no work.
+   group("default") {
+     testonly = true
+-    deps = [ ":webrtc" ]
++    deps = [ ":webrtc","//libwebrtc:libwebrtc", ]
+     if (rtc_build_examples) {
+       deps += [ "examples" ]
+     }
diff --git a/run.py b/run.py
index 4996dbd..05a653a 100644
--- a/run.py
+++ b/run.py
@@ -163,85 +163,97 @@ def get_depot_tools(source_dir, fetch=False):
 
 
 PATCH_INFO = {
-    '4k.patch': (2, []),
-    'macos_h264_encoder.patch': (2, []),
-    'macos_screen_capture.patch': (2, []),
+    'libwebrtc': (1, []),
+    # '4k.patch': (2, []),
+    # 'macos_h264_encoder.patch': (2, []),
+    # 'macos_screen_capture.patch': (2, []),
     'ios_bitcode.patch': (1, ['build']),
     'ios_disable_iossim.patch': (1, ['build']),
 }
 
 PATCHES = {
     'windows': [
-        '4k.patch',
-        'windows_add_deps.patch',
-        'ssl_verify_callback_with_native_handle.patch',
+        'libwebrtc',
+        # '4k.patch',
+        # 'windows_add_deps.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
     ],
     'macos_x86_64': [
-        'add_dep_zlib.patch',
-        '4k.patch',
-        'macos_h264_encoder.patch',
-        'macos_screen_capture.patch',
-        'macos_simulcast.patch',
-        'ios_simulcast.patch',
-        'ssl_verify_callback_with_native_handle.patch',
+        'libwebrtc',
+        # 'add_dep_zlib.patch',
+        # '4k.patch',
+        # 'macos_h264_encoder.patch',
+        # 'macos_screen_capture.patch',
+        # 'macos_simulcast.patch',
+        # 'ios_simulcast.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
     ],
     'macos_arm64': [
-        'add_dep_zlib.patch',
-        '4k.patch',
-        'macos_h264_encoder.patch',
-        'macos_screen_capture.patch',
-        'macos_simulcast.patch',
-        'ios_simulcast.patch',
-        'ssl_verify_callback_with_native_handle.patch',
+        'libwebrtc',
+        # 'add_dep_zlib.patch',
+        # '4k.patch',
+        # 'macos_h264_encoder.patch',
+        # 'macos_screen_capture.patch',
+        # 'macos_simulcast.patch',
+        # 'ios_simulcast.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
     ],
     'ios': [
-        'add_dep_zlib.patch',
-        '4k.patch',
-        'macos_h264_encoder.patch',
-        'macos_screen_capture.patch',
-        'macos_simulcast.patch',
-        'ios_manual_audio_input.patch',
-        'ios_simulcast.patch',
-        'ssl_verify_callback_with_native_handle.patch',
-        'ios_bitcode.patch',
+        'libwebrtc',
+        # 'add_dep_zlib.patch',
+        # '4k.patch',
+        # 'macos_h264_encoder.patch',
+        # 'macos_screen_capture.patch',
+        # 'macos_simulcast.patch',
+        # 'ios_manual_audio_input.patch',
+        # 'ios_simulcast.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
+        # 'ios_bitcode.patch',
         'ios_disable_iossim.patch',
     ],
     'android': [
-        'add_dep_zlib.patch',
-        '4k.patch',
-        'ssl_verify_callback_with_native_handle.patch',
-        'android_webrtc_version.patch',
-        'android_fixsegv.patch',
-        'android_simulcast.patch',
+        'libwebrtc',
+        # 'add_dep_zlib.patch',
+        # '4k.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
+        # 'android_webrtc_version.patch',
+        # 'android_fixsegv.patch',
+        # 'android_simulcast.patch',
     ],
     'raspberry-pi-os_armv6': [
-        'nacl_armv6_2.patch',
-        'add_dep_zlib.patch',
-        '4k.patch',
-        'ssl_verify_callback_with_native_handle.patch',
+        'libwebrtc',
+        # 'nacl_armv6_2.patch',
+        # 'add_dep_zlib.patch',
+        # '4k.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
     ],
     'raspberry-pi-os_armv7': [
-        'add_dep_zlib.patch',
-        '4k.patch',
-        'ssl_verify_callback_with_native_handle.patch',
+        'libwebrtc',
+        # 'add_dep_zlib.patch',
+        # '4k.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
     ],
     'raspberry-pi-os_armv8': [
-        'add_dep_zlib.patch',
-        '4k.patch',
-        'ssl_verify_callback_with_native_handle.patch',
+        'libwebrtc',
+        # 'add_dep_zlib.patch',
+        # '4k.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
     ],
     'ubuntu-18.04_armv8': [
-        'add_dep_zlib.patch',
-        '4k.patch',
-        'ssl_verify_callback_with_native_handle.patch',
+        'libwebrtc',
+        # 'add_dep_zlib.patch',
+        # '4k.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
     ],
     'ubuntu-18.04_x86_64': [
-        '4k.patch',
-        'ssl_verify_callback_with_native_handle.patch',
+        'libwebrtc',
+        # '4k.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
     ],
     'ubuntu-20.04_x86_64': [
-        '4k.patch',
-        'ssl_verify_callback_with_native_handle.patch',
+        'libwebrtc',
+        # '4k.patch',
+        # 'ssl_verify_callback_with_native_handle.patch',
     ]
 }
 
@@ -285,6 +297,7 @@ def get_webrtc(source_dir, patch_dir, version, target,
             cmd(['git', 'checkout', '-f', version])
             cmd(['git', 'clean', '-df'])
             cmd(['gclient', 'sync', '-D', '--force', '--reset', '--with_branch_heads'])
+            cmd(['git', 'clone', '--depth', '1', 'https://github.com/webrtc-sdk/libwebrtc'])
             for patch in PATCHES[target]:
                 depth, dirs = PATCH_INFO.get(patch, (1, ['.']))
                 dir = os.path.join(src_dir, *dirs)
